src/main/java/net/tropicraft/core/common/dimension/mapgen/MapGenVolcano.java

```diff
@@ -0,0 +1,260 @@
+package net.tropicraft.core.common.dimension.mapgen;
+
+import net.minecraft.block.Block;
+import net.minecraft.block.BlockState;
+import net.minecraft.block.Blocks;
+import net.minecraft.block.material.Material;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.world.IWorld;
+import net.minecraft.world.World;
+import net.minecraft.world.biome.Biome;
+import net.minecraft.world.biome.provider.BiomeProvider;
+import net.minecraft.world.chunk.ChunkPrimer;
+import net.tropicraft.core.common.block.TropicraftBlocks;
+import net.tropicraft.core.common.dimension.biome.TropicraftBiomes;
+import net.tropicraft.core.common.dimension.noise.NoiseModule;
+import net.tropicraft.core.common.dimension.noise.generator.Billowed;
+
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Random;
+
+public class MapGenVolcano {
```

Comment:
"MapGen" is an outdated term, this could just be "VolcanoGenerator" or something like that